nimbus-primitives/src/lib.rs

```diff
@@ -35,6 +36,19 @@ pub use digests::CompatibleDigestItem;
 
 pub use inherents::{InherentDataProvider, INHERENT_IDENTIFIER};
 
+/// Input NimbusId and returns the additional inherent digests
+pub trait InherentDigestsProvider<Id> {
+	type Digests: IntoIterator<Item = DigestItem>;
+	fn provide_inherent_digests(&self, id: Id) -> Self::Digests;
```

Comment:
It would be better to also add the parent hash as a parameter, to allow the implementation to query the storage if needed.

Reply:
Can this be done without making the function (and trait) async?